src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2880,6 +2880,186 @@ export default class EthereumApi implements Api {
+   * @returns transaction base fee per gas and effective priority fee per gas for the requested/supported block range
+   *
+   * * `oldestBlock`:  - Lowest number block of the returned range.
+   * * `baseFeePerGas`:  - An array of block base fees per gas. This includes the next block after the newest of the returned range,
+   * because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
+   * * `gasUsedRatio`:  - An array of block gas used ratios. These are calculated as the ratio of `gasUsed` and `gasLimit`.
+   * * `reward`:  - (optional) An array of effective priority fee per gas data points from a single block. All zeroes are returned if the
+   * block is empty.
+   *
+   * @example
+   * ```javascript
+   * const feeHistory = await provider.request({ method: "eth_feeHistory", params: ["0x5", "0xE5A51D", [10, 50, 80]] });
+   * console.log(feeHistory);
+   * ```
+   */
+  @assertArgLength(3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles: number[]
+  ): Promise<Ethereum.FeeHistory<"private">> {
+    const blockchain = this.#blockchain;
+    const PAD_PRECISION = 16;
+    const PRECISION_FLOAT = 1e14;
+    const PRECISION_BIG_INT = BigInt(1e16);
+
+    const newestBlockNumber = blockchain.blocks
+      .getEffectiveNumber(newestBlock)
+      .toNumber();
+
+    // blockCount > newestBlock is technically valid but we cannot go past the Genesis Block.
+    const totalBlocks = Math.min(
+      Quantity.toNumber(blockCount),
+      newestBlockNumber + 1
+    );
+
+    // Cut out early if no range is given.
+    if (totalBlocks === 0) {
+      return {
+        oldestBlock: Quantity.from(newestBlockNumber),
+        baseFeePerGas: undefined,
+        gasUsedRatio: null,
+        reward: undefined
+      };
+    }
+
+    const baseFeePerGas = new Array(totalBlocks);
+    const gasUsedRatio = new Array(totalBlocks);
+    let reward;
+    if (rewardPercentiles.length > 0) {
+      reward = new Array(totalBlocks);
+    }
+    // blockCount is inclusive of newestBlock
+    const oldestBlockNumber = newestBlockNumber - (totalBlocks - 1);
+    let currentBlockNumber = oldestBlockNumber;
+    let currentBlock: Block;
+
+    while (currentBlockNumber <= newestBlockNumber) {
+      currentBlock = await blockchain.blocks.get(
+        Quantity.toBuffer(currentBlockNumber)
+      );
+
+      const currentPosition = currentBlockNumber - oldestBlockNumber;
+
+      baseFeePerGas[currentPosition] =
+        currentBlock.header.baseFeePerGas || Quantity.Zero;
+
+      const { gasUsed, gasLimit } = currentBlock.header;
+      if (gasUsed === gasLimit) {
+        gasUsedRatio[currentPosition] = 1;
+      } else {
+        gasUsedRatio[currentPosition] = Number(
+          `0.${(
+            (currentBlock.header.gasUsed.toBigInt() * PRECISION_BIG_INT) /
+            currentBlock.header.gasLimit.toBigInt()
+          )
```

Comment:
Since we've already pulled out `gasUsed` and `gasLimit` from the header let's just use those:

```suggestion
          `0.${((gasUsed.toBigInt() * PRECISION_BIG_INT) / gasLimit.toBigInt())
```

(You might want to commit this one yourself instead of using the suggestion above, since the formatting could be different)

Reply:
:+1: